### PR TITLE
Feature step4 integration

### PR DESCRIPTION
**Pull request by kmilo9999**
- Added A02c_IOWAGA_thredds_prior.py
- Added new start up configuration
- Fixing dependencies 
- Clearing code and formatting file

### CODE REVIEW

**Comment by cpaniaguam** [comment]
@kmilo9999 Out of the three files changed in this PR, two are being modified in #45; the only new one is `src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py`. This might lead to merge conflicts later. Should we wait until #45 is merged before reviewing this one?

**Review by cpaniaguam — COMMENTED** [review]
I added a few commits. Awating for @hollandjg for review.

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+import datetime
+
+import h5py
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.font_manager as font_for_print
+from matplotlib.gridspec import GridSpec
+import xarray as xr
+from siphon.catalog import TDSCatalog
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:11]: @kmilo9999 It seems like this `siphon` dep might be missing: https://github.com/brown-ccv/icesat2-tracks/actions/runs/7557763959/job/20577748725?pr=50

Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:10]: It was also missed from the `pyprojet.toml`. I added it too.

**Review by cpaniaguam — APPROVED** [review]
👍

**Pull request merged**